screens/ChooseLocationScreen.tsx

```diff
@@ -0,0 +1,274 @@
+import * as LocationProvider from 'expo-location';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { Image, StyleSheet, TouchableOpacity, View } from 'react-native';
+import MapView, { EdgeInsets, Marker, Region } from 'react-native-maps';
+import {
+  ActivityIndicator,
+  Caption,
+  FAB,
+  TextInput,
+  useTheme,
+} from 'react-native-paper';
+import { Theme } from 'react-native-paper/lib/typescript/types';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { useNavigation } from '@react-navigation/native';
+import currentLocationIcon from '../assets/current-location.png';
+import { SecondaryAppbar } from '../components/SecondaryAppbar';
+import { MapContext } from '../context';
+import { findCurrentLocation, findPlace } from '../context/MapContext';
+import { getHotspotMarker } from '../models/Hotspot';
+import {
+  defaultLocation,
+  getFormattedAddress,
+  Location,
+} from '../models/Location';
+import { RootStackScreenProps } from '../types';
+import { loadHotspots } from '../utils/hotspots';
+
+const intialRegion: Region = {
+  latitude: 46.77293258116839,
+  longitude: 23.587688864363546,
+  latitudeDelta: 0.0922,
+  longitudeDelta: 0.0421,
+};
```

Comment:
the initial region should be the region displayed in the previous map screen

Reply:
done.